assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -5078,6 +5046,17 @@ else if ("remove".equals(addUpdateRemoveAssignment)) {
             }
         }
     } // integrateGradebook
+    
+    private Stream<User> getSubmitters(final AssignmentSubmission aSubmission, final String method) {
+		return aSubmission.getSubmitters().stream().map(u -> {
+		    try {
+		        return userDirectoryService.getUser(u.getSubmitter());
```

Comment:
getUser is not optimized. This should be re-worked to use getUsers .... this could potentially save hundreds of individual LDAP calls in a large site.

Reply:
It looks like https://jira.sakaiproject.org/browse/SAK-33927 introduced this getUser pattern in the first place. Problem of using these fancy streams. Consolidating all of this makes it fixable in one place though. (Generate a Collection<String> in here and pass it afterward.

Reply:
Its probably more accurate to say getUser is not optimized when going to a provider over getUsers. Both methods use the same cache so subsequent calls are not an issue.